app/src/org/commcare/fragments/connect/ConnectJobsListsFragment.java

```diff
@@ -125,8 +125,16 @@ private void launchAppForJob(ConnectJobRecord job, boolean isLearning) {
         setActiveJob(job);
 
         String appId = isLearning ? job.getLearnAppInfo().getAppId() : job.getDeliveryAppInfo().getAppId();
+        boolean appInstalled = AppUtils.isAppInstalled(appId);
 
-        if (AppUtils.isAppInstalled(appId)) {
+        // We need the composite job to get the correct number of deliveries.
+        ConnectJobRecord compositeJob = ConnectJobUtils.getCompositeJob(requireActivity(), job.getJobId());
+        boolean deliveryComplete = compositeJob != null &&
+                (compositeJob.isFinished() || compositeJob.getDeliveries().size() >= compositeJob.getMaxVisits());
+
+        if (appInstalled && deliveryComplete && !isLearning) {
+            ConnectAppUtils.INSTANCE.launchAppForCompletedDelivery(requireActivity(), appId, job);
```

Comment:
Think instead all we want here is to open the delivery progress fragment by defining a new action from jobs list fragment to the delivery progress fragment in the corresponding nav_graph_connect.xml, that will also preserve the back behaviour so that user gets back to the jobs list screen when they click back.